core/src/main/java/com/turn/ttorrent/client/SharedTorrent.java

```diff
@@ -257,7 +252,7 @@ public SharedTorrent(byte[] torrent, File parent, boolean seeder,
 
 		this.initialized = false;
 		this.pieces = new Piece[0];
-		this.rarest = Collections.synchronizedSortedSet(new TreeSet<Piece>());
+		this.rarest = new HashSet<Integer>();
```

Comment:
Oh, don't you also need to make this a synchronized set?

Reply:
It is not needed, because `rarest` field is always used in `synchronized` functions